### PR TITLE
Add `webgpu` IDL spec and generated code

### PR DESCRIPTION
**Pull request by MaxDesiatov**
This allows removing manual `OffscreenRenderingContext` and `RenderingContext` declarations.

### CODE REVIEW

**Comment by j-f1** [comment]
Looks like this needs to be rebased?

**Comment by MaxDesiatov** [comment]
I've rebased it, but there are still a few commits that don't matter in the end diff, and they will be squashed. Yes, the diff on `Generated.swift` is huge, because `webgpu` is apparently that big...

File: Sources/WebIDL/Declaration.swift [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /// https://github.com/w3c/webidl2.js#iterable-async-iterable-maplike-and-setlike-declarations
-public protocol IDLDeclaration: IDLNode, IDLInterfaceMember {
+public protocol IDLDeclaration: IDLInterfaceMember {
```
Comment (MaxDesiatov) [inline Sources/WebIDL/Declaration.swift:2]: This conformance to `IDLNode` triggered a redundancy warning, as apparently `IDLInterfaceMember` already conforms to `IDLNode`.

File: Sources/DOMKitDemo/WebGLDemo.swift [hunk 2]
```diff
@@ -117,7 +117,7 @@ func runWebGLDemo() {
   // Get A WebGL context
   let canvas = HTMLCanvasElement(from: document.createElement(localName: "canvas"))!
   _ = document.body?.appendChild(node: canvas)
-  let context = canvas.getContext(contextId: "webgl2")!.webGL2RenderingContext!
+  let context = WebGL2RenderingContext.construct(from: canvas.getContext(contextId: "webgl2")!.jsValue)!
```
Comment (j-f1) [inline Sources/DOMKitDemo/WebGLDemo.swift:120]:
```suggestion
  let context = canvas.getContext(contextId: "webgl2").webGL2RenderingContext!
```

**Review by j-f1 — APPROVED** [review]

File: Sources/DOMKitDemo/WebGLDemo.swift (hunk 2 above)
Comment (MaxDesiatov) [reply Sources/DOMKitDemo/WebGLDemo.swift:120]: This doesn't work unfortunately, as these enum properties aren't public in the generated code. IDK if that's on purpose. Either way, I decided to make it work with the existing generation approach, and those properties could be made public, if needed, in a separate PR.

Comment (j-f1) [reply Sources/DOMKitDemo/WebGLDemo.swift:120]: Oh they should definitely be public — there’s no internal use for them.

**Pull request merged**